rust/noosphere-gateway/src/worker/name_system.rs

```diff
@@ -232,8 +232,13 @@ where
                 context,
                 republish,
             } => {
-                if let Err(error) = set_counterpart_record(context, &record).await {
-                    warn!("Could not set counterpart record on sphere: {error}");
+                // NOTE: Very important not to update this record on every
+                // re-publish, otherwise we will generate a new sphere revision
+                // on an on-going basis indefinitely
+                if !republish {
```

Comment:
:exploding_head: great catch. the test here ensures we handle republishing w/r/t expiry, but does not consider the effects on setting counterpart. Might be able to easily extend the test and check counterpart record